satpy/readers/utils.py

```diff
@@ -101,12 +102,12 @@ def get_geostationary_mask(area):
 
 def _lonlat_from_geos_angle(x, y, geos_area):
     """Get lons and lats from x, y in projection coordinates."""
-    h = (geos_area.proj_dict['h'] + geos_area.proj_dict['a']) / 1000
+    h = (geos_area.proj_dict['h'] + geos_area.proj_dict['a']) / 1000.0
     b__ = (geos_area.proj_dict['a'] / geos_area.proj_dict['b']) ** 2
```

Comment:
Shouldn't `b` be converted to float` here?

Reply:
Good point